packages/craco/lib/features/webpack/eslint.js

```diff
@@ -48,80 +48,132 @@ function extendsEslintConfig(loader, eslintConfig, context) {
     }
 }
 
-function useEslintConfigFile(loader) {
-    if (loader.options) {
-        loader.options.useEslintrc = true;
-        delete loader.options.baseConfig;
+function useEslintConfigFile(webpackEslintConfig) {
+    if (webpackEslintConfig.options) {
+        webpackEslintConfig.options.useEslintrc = true;
+        delete webpackEslintConfig.options.baseConfig;
     } else {
-        loader.options = {
+        webpackEslintConfig.options = {
             useEslintrc: true
         };
     }
 
     log("Overrided ESLint config to use a config file.");
 }
 
-function enableEslintIgnoreFile(loader) {
-    if (loader.options) {
-        loader.options.ignore = true;
+function enableEslintIgnoreFile(webpackEslintConfig) {
+    if (webpackEslintConfig.options) {
+        webpackEslintConfig.options.ignore = true;
     } else {
-        loader.options = {
+        webpackEslintConfig.options = {
             ignore: true
         };
     }
 
     log("Overrided ESLint config to enable an ignore file.");
 }
 
-function applyLoaderOptions(loader, loaderOptions, context) {
+function applyLoaderOptions(webpackEslintConfig, loaderOptions, context) {
```

Comment:
I guess now the options are not only for the "loaders" but also for the "plugins" ?

If so, we should rename the function.

Does the loader and the plugin support the same options? Or at the least the options object have the same structure?

Reply:
Both seems to use https://eslint.org/docs/developer-guide/nodejs-api#%E2%97%86-new-eslint-options, so the options are very similar.

As of renaming loaderOptions to options, it's all good with me but it would cause a breaking change for all craco configs since instead of 

`

    eslint: {
        enable: true /* (default value) */,
        mode: "extends" /* (default value) */ || "file",
        configure: { /* Any eslint configuration options: https://eslint.org/docs/user-guide/configuring */ },
        configure: (eslintConfig, { env, paths }) => { return eslintConfig; },
        loaderOptions: { /* Any eslint-loader configuration options: https://github.com/webpack-contrib/eslint-loader. */ },
        loaderOptions: (eslintOptions, { env, paths }) => { return eslintOptions; }
    },
`
It would be
`

    eslint: {
        enable: true /* (default value) */,
        mode: "extends" /* (default value) */ || "file",
        configure: { /* Any eslint configuration options: https://eslint.org/docs/user-guide/configuring */ },
        configure: (eslintConfig, { env, paths }) => { return eslintConfig; },
        options: { /* Any eslint-loader configuration options: https://eslint.org/docs/developer-guide/nodejs-api#-new-eslintoptions. */ },
        options: (eslintOptions, { env, paths }) => { return eslintOptions; }
    },
`

Reply:
I agree, it will be a major release with a migration path!